File: engine/model/src/main/scala/tools/packet/monster/DamageMonster.scala
package tools.packet.monster

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class DamageMonster(private var _objectId: Int, private var _damage: Int) extends PacketInput {
  def objectId: Int = _objectId

  def damage: Int = _damage

  override def opcode(): SendOpcode = SendOpcode.DAMAGE_MONSTER
}

File: engine/engine-model/src/main/java/tools/packet/monster/DamageMonster.java
package tools.packet.monster;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record DamageMonster(Integer objectId, Integer damage) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.DAMAGE_MONSTER;
   }
}